### PR TITLE
[WIP] Game state refactor

### PR DESCRIPTION
**Pull request by hackedhead**
Don't merge this yet, but here's a PR so you can see where I'm going.

Ideally, Dungeon will handle the entire game state, and we can write a blessed layer that consumes it and draws the GUI after each state change. I'm also planning to add logging, which we can leverage for the history readout ("Entered a new room", "equipped a shield with power X", "drank a potion", "gained Y health", "fought a monster of power Z", "took Q damage", etc).

### CODE REVIEW

**Comment by setphen** [comment]
These changes look good. I'll test these and get to work on re-writing the drawing portion.

File: game/models.py [hunk 1]
```diff
@@ -172,17 +177,18 @@ def make_standard_deck():
 class Dungeon:
     """Handle deck, room and player creation and interaction"""
 
-    def __init__(self, cards=make_standard_deck(), seed=None):
+    def __init__(self, cards=None, seed=None):
```
Comment (setphen) [inline game/models.py]: Like this idea of passing a set of cards to the dungeon. I guess you could recreate a certain point in a dungeon from a game history and see if a game was winnable from a certain point.

Which makes me think we should store the state of the deck in the game history, in addition to the console output.

Comment (hackedhead) [reply game/models.py]: well, the dungeon progresses through the deck in a certain way based on the PRNG, so to recreate a gamestate, you should feed it the same deck (set of cards), same seed, and then the same (partial) sequence of inputs. But yes, this design would let you do this rapidly (save inputs and replay them "behind the scenes").

The idea behind taking the card set as a variable is to make it easier to test different decks (different values for face cards, different amounts of cards (two decks together), etc)

Comment (setphen) [reply game/models.py]: Ah, right. That makes a lot more sense.

**Pull request merged**